ide/extexecution.base/src/org/netbeans/api/extexecution/base/ExplicitProcessParameters.java

```diff
@@ -0,0 +1,444 @@
+        
+        private void initArgs() {
+            if (arguments == null) {
+                arguments = new ArrayList<>();
+                /*
+                if (appendArgs == null) {
+                    appendArgs = false;
+                }
+                */
+            }
+        }
+        
+        /**
+         * Appends a single argument. {@code null} is ignored.
+         * @param a argument
+         * @return the builder
+         */
+        public Builder arg(@NullAllowed String a) {
+            if (a == null) {
+                return this;
+            }
+            initArgs();
+            arguments.add(a);
+            return this;
+        }
+
+        /**
+         * Appends arguments in the list. {@code null} is ignored as well as {@code null}
+         * items in the list.
+         * @param args argument list
+         * @return the builder
+         */
+        public Builder args(@NullAllowed List<String> args) {
+            if (args == null) {
+                return this;
+            }
+            // init even if the list is empty.
+            initArgs();
+            args.forEach(this::arg);
+            return this;
+        }
+
+        /**
+         * Appends arguments in the list. {@code null} is ignored as well as {@code null}
+         * items in the list.
+         * @param args argument list
+         * @return the builder
+         */
+        public Builder args(@NullAllowed String... args) {
+            if (args == null) {
+                return this;
+            }
+            return args(Arrays.asList(args));
+        }
+        
+        private void initPriorityArgs() {
+            if (priorityArguments == null) {
+                priorityArguments = new ArrayList<>();
+                /*
+                if (appendPriorityArgs == null) {
+                    appendPriorityArgs = true;
+                }
+                */
+            }
+        }
+        
+        /**
+         * Appends a single priority argument. {@code null} is ignored.
+         * @param a priority argument
+         * @return the builder
+         */
+        public Builder priorityArg(@NullAllowed String a) {
+            if (a == null) {
+                return this;
+            }
+            initPriorityArgs();
+            priorityArguments.add(a);
+            return this;
+        }
+
+        /**
+         * Appends arguments in the list. {@code null} is ignored as well as {@code null}
+         * items in the list.
+         * @param args argument list
+         * @return the builder
+         */
+        public Builder priorityArgs(@NullAllowed List<String> args) {
+            if (args == null) {
+                return this;
+            }
+            initPriorityArgs();
+            args.forEach(this::priorityArg);
+            return this;
+        }
+
+        /**
+         * Appends arguments in the list. {@code null} is ignored as well as {@code null}
```

Comment:
Appends!? There is also `appendXYZ` method next to here and that one doesn't append! That one is changing the replacing mode. Consider use "replace" terminology instead.

Reply:
OK, the terminology sins start to pile up. So what about to:
- rename `priorityArgs` -> `launcherArgs`. VM arguments, launcher script arguments, ...
- rename `append(Launcher)Args` -> `replace(Launcher)Args` + invert the logic (true -> replace)
- rename `rank` to `position` (better than priority, if sorted lowest-first)

Reply:
Done in 600178d5af